lib/ae_mdw_web/plugs/request_span.ex

```diff
@@ -0,0 +1,34 @@
+defmodule AeMdwWeb.Plugs.RequestSpan do
+  @moduledoc """
+  Emits telemetry request duration events associating them to a route.
+  """
+
+  alias Plug.Conn
+
+  @type opts() :: []
+
+  @spec init(opts()) :: opts()
+  def init(opts), do: opts
+
+  @spec call(Conn.t(), opts()) :: Conn.t()
+  def call(conn, _opts) do
+    conn
+    |> Conn.assign(:start_time, System.monotonic_time(:millisecond))
+    |> Conn.register_before_send(&emit_event_callback/1)
+  end
+
+  @spec emit_event_callback(Conn.t()) :: Conn.t()
+  def emit_event_callback(conn) do
+    duration = System.monotonic_time(:millisecond) - conn.assigns.start_time
+
+    with %{route: route} <-
+           Phoenix.Router.route_info(AeMdwWeb.Router, conn.method, conn.request_path, conn.host),
```

Comment:
Can we pattern-match these out of the `Conn` struct? That way we get compiler errors if there's any typo

Reply:
Sure, tests would fail but changed to pattern matching style.